cli/dts/lib.esnext.array.d.ts

```diff
@@ -38,6 +38,38 @@ interface Array<T> {
   * predicate. If it is not provided, undefined is used instead.
   */
    findLastIndex(predicate: (value: T, index: number, obj: T[]) => unknown, thisArg?: any): number;
+
+  /** Copies and reverses the elements in an array.*/
+  toReversed(): T[];
+
+  /**
+   * Copies and sorts an array.
+   * @param compareFn Function used to determine the order of the elements. It is expected to return
+   * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
+   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * ```ts
+   * [11,2,22,1].sort((a, b) => a - b)
+   * ```
+   */
+  toSorted(compareFn?: (a: T, b: T) => number): T[];
+
+  /**
+   * Copies and elements from an array and, if necessary, inserts new elements in their place, returning the deleted elements.
+   * @param start The zero-based location in the array from which to start removing elements.
+   * @param deleteCount The number of elements to remove.
+   * @returns An array containing the elements that were deleted.
+   */
+  toSpliced(start: number, deleteCount?: number): T[];
+  /**
+   * Copies and removes elements from an array and, if necessary, inserts new elements in their place, returning the deleted elements.
+   * @param start The zero-based location in the array from which to start removing elements.
+   * @param deleteCount The number of elements to remove.
+   * @param items Elements to insert into the array in place of the deleted elements.
+   * @returns An array containing the elements that were deleted.
+   */
+  toSpliced(start: number, deleteCount: number, ...items: T[]): T[];
+
+  with(index: number, value: any): any[];
```

Comment:
```suggestion
  with<F>(index: number, value: F): (T | F)[];
```
Many different options here, but I'd much rather have a stricter definition first instead of breaking code later.

Reply:
built-in typings arent ever complex typings like that, so we will leave it to `any`.

Reply:
I'm not sure if it's that simple why there aren't complex typings.  Many of the array methods were implemented in far older type systems so they did not have the ability to use these new type features without breaking old code.  I'd consider this an inflection point in improving the user-experience with types.  Also, if you type this as `any`, then no one who cares about sufficiently safe code will want to use this new feature.  Please reconsider.